src/test/java/org/openrewrite/java/micronaut/AddMicronautRetryDependencyIfNeededTest.java

```diff
@@ -0,0 +1,126 @@
+    }
+
+    @Language("java")
+    private final String retryableService = """
+            import jakarta.inject.Singleton;
+            import io.micronaut.retry.annotation.Retryable;
+            
+            @Singleton
+            public class PersonService {
+            
+                @Retryable
+                public void callFlakyRemote() {
+                    
+                }
+            }
+        """;
+
+    @Language("groovy")
+    private final String buildGradleInitial = """
+            plugins {
+                id("io.micronaut.application") version "4.0.0-M2"
+            }
+            
+            micronaut { version '4.0.0-M2'}
+            
+            repositories {
+                mavenCentral()
+            }
+        """;
+
+    @Language("groovy")
+    private final String buildGradleExpected = """
+            plugins {
+                id("io.micronaut.application") version "4.0.0-M2"
+            }
+            
+            micronaut { version '4.0.0-M2'}
+            
+            repositories {
+                mavenCentral()
+            }
+            
+            dependencies {
+                implementation "io.micronaut:micronaut-retry"
+            }
+        """;
+
+    @Language("xml")
+    private final String pomInitial = """
+            <project>
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                <parent>
+                    <groupId>io.micronaut.platform</groupId>
+                    <artifactId>micronaut-parent</artifactId>
+                    <version>4.0.0-M2</version>
+                </parent>
+            </project>
+        """;
+
+    @Language("xml")
+    private final String pomExpected = """
+            <project>
+                <groupId>com.mycompany.app</groupId>
+                <artifactId>my-app</artifactId>
+                <version>1</version>
+                <parent>
+                    <groupId>io.micronaut.platform</groupId>
+                    <artifactId>micronaut-parent</artifactId>
+                    <version>4.0.0-M2</version>
+                </parent>
+                <dependencies>
+                    <dependency>
+                        <groupId>io.micronaut</groupId>
+                        <artifactId>micronaut-retry</artifactId>
+                    </dependency>
+                </dependencies>
+            </project>
+        """;
+
+    @Test
+    void updateGradleDependencies() {
+        rewriteRun(spec -> spec.beforeRecipe(withToolingApi()), mavenProject("project", srcMainJava(java(retryableService)), buildGradle(buildGradleInitial, buildGradleExpected)));
+    }
+
+    @Test
+    void updateMavenDependencies() {
+        rewriteRun(mavenProject("project", srcMainJava(java(retryableService)), pomXml(pomInitial, pomExpected)));
+    }
```

Comment:
I'd not yet seen this style of having the code blocks separate from the tests that use them. Which is fine; I kind of like this as well for a change, but know that is different from what we typically have: test blocks directly within the `rewriteRun` invocation, which can save some lookups when reading the tests.

Reply:
Yes, I've noticed that in looking at the other repos. This style here resulted from not wanting to repeat the same code block in the Gradle and Maven variations of the test methods. It would be quite a lot of repetition in `AddSnakeYamlDependencyIfNeededTest`, for example. For these smaller ones, I suppose it might make sense to declare the non-repeated blocks as local variables directly in the test method at the very least. I like keeping them as separate variables so that I still get the syntax highlighting from the `Language` annotation.

Reply:
Ah yes for the code highlighting we've found comments tend to work there.. so for instance `//language=java` just before a `java(...)` invocation. If that helps at all.

Reply:
Ahh, interesting, I didn't know that worked.